File: email-service/src/main/java/com/rj/ecommerce_email_service/messaging/admin/QueueAdminController.java
package com.rj.ecommerce_email_service.messaging.admin;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.amqp.rabbit.core.RabbitAdmin;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.rj.ecommerce_email_service.messaging.config.RabbitMQConfig;
import com.rj.ecommerce_email_service.messaging.listener.RabbitMQDeadLetterConfig;

@RestController
@RequestMapping("/admin/queues")
@RequiredArgsConstructor
@Slf4j
public class QueueAdminController {

    private final RabbitAdmin rabbitAdmin;

    @DeleteMapping("/purge/{queueName}")
    public ResponseEntity<String> purgeQueue(@PathVariable String queueName) {
        log.info("Purging queue: {}", queueName);
        
        String actualQueueName;
        
        // Map friendly names to actual queue names
        switch (queueName) {
            case "email":
                actualQueueName = RabbitMQConfig.EMAIL_QUEUE;
                break;
            case "dlq":
                actualQueueName = RabbitMQDeadLetterConfig.EMAIL_DLQ;
                break;
            case "notification":
                actualQueueName = RabbitMQConfig.NOTIFICATION_QUEUE;
                break;
            default:
                actualQueueName = queueName;
        }
        
        rabbitAdmin.purgeQueue(actualQueueName, false);
        
        return ResponseEntity.ok("Queue " + actualQueueName + " has been purged");
    }
    
    @DeleteMapping("/purge-all")
    public ResponseEntity<String> purgeAllQueues() {
        log.info("Purging all queues");
        
        // Purge main queues
        rabbitAdmin.purgeQueue(RabbitMQConfig.EMAIL_QUEUE, false);
        rabbitAdmin.purgeQueue(RabbitMQConfig.NOTIFICATION_QUEUE, false);
        
        // Purge dead letter queue
        rabbitAdmin.purgeQueue(RabbitMQDeadLetterConfig.EMAIL_DLQ, false);
        
        return ResponseEntity.ok("All queues have been purged");
    }
}

File: notification-service/src/main/kotlin/com/rj/notification_service/messaging/admin/QueueAdminController.kt
// File: email-service/src/main/kotlin/com/rj/ecommerce_email_service/messaging/admin/QueueAdminController.kt
package com.rj.notification_service.messaging.admin

import com.rj.notification_service.config.AppProperties
import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.amqp.rabbit.core.RabbitAdmin
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

private val log = KotlinLogging.logger {}

@RestController
@RequestMapping("/admin/queues")
class QueueAdminController(
    private val rabbitAdmin: RabbitAdmin,
    private val appProperties: AppProperties
) {

    @DeleteMapping("/purge/{queueName}")
    fun purgeQueue(@PathVariable queueName: String): ResponseEntity<String> {
        log.warn { "Received admin request to purge queue: '$queueName'" }

        // Use the type-safe properties to find the actual queue name.
        val queueToPurge = when (queueName.lowercase()) {
            "email" -> appProperties.rabbitmq.notificationRequest.queue
            "dlq" -> "${appProperties.rabbitmq.notificationRequest.queue}.dlq"
            "receipt" -> appProperties.rabbitmq.notificationReceipt.queue
            else -> {
                log.error { "Attempted to purge an unknown queue alias: '$queueName'" }
                return ResponseEntity.badRequest().body("Unknown queue alias: '$queueName'. Known aliases: email, dlq, receipt.")
            }
        }

        val messageCount = rabbitAdmin.purgeQueue(queueToPurge, false)
        val responseMessage = "Queue '$queueToPurge' has been purged. $messageCount messages were deleted."
        log.info { responseMessage }

        return ResponseEntity.ok(responseMessage)
    }

    @DeleteMapping("/purge-all")
    fun purgeAllQueues(): ResponseEntity<String> {
        log.warn { "Received admin request to purge ALL known queues." }

        val emailQueue = appProperties.rabbitmq.notificationRequest.queue
        val dlq = "$emailQueue.dlq"
        val receiptQueue = appProperties.rabbitmq.notificationReceipt.queue

        rabbitAdmin.purgeQueue(emailQueue)
        rabbitAdmin.purgeQueue(dlq)
        rabbitAdmin.purgeQueue(receiptQueue)

        val responseMessage = "All known queues ($emailQueue, $dlq, $receiptQueue) have been purged."
        log.info { responseMessage }

        return ResponseEntity.ok(responseMessage)
    }
}